### PR TITLE
Bump up Gravatar to 2.0.0

### PR DESCRIPTION
**Pull request by pinarol**
Fixes #

[Gravatar SDK](https://github.com/Automattic/Gravatar-SDK-iOS) has a new release(yaay 🎉 ). This PR makes necessary updates to use it.

To test:

Smoke test scenarios in https://github.com/wordpress-mobile/WordPress-iOS/issues/22543. The avatars should load fine. Uploading a new avatar should also work fine.

## Regression Notes
1. Potential unintended areas of impact

All avatars in the app.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I documented the use cases in https://github.com/wordpress-mobile/WordPress-iOS/issues/22543 and tested those.

4. What automated tests I added (or what prevented me from doing so)

No new automated tests were added since we aren't adding new functionality.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [x] WordPress.com sites and self-hosted Jetpack sites.
- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] VoiceOver.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] iPhone and iPad. 
- [x] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Gravatar/UIImageView+Gravatar.swift [hunk 1]
```diff
@@ -58,26 +58,26 @@ extension UIImageView {
...
+                options = [.imageCache(cache)]
+            }
+            else {
+                // If we don't pass any cache to `gravatar.setImage(...)` it will use its internal in-memory cache.
+                // But in order to make things work consistently with the rest of the app, we should use `WordPressUI.ImageCache.shared` here.
+                // It needs to be fixed if we fail to do that. That's why we put an assertionFailure here.
+                assertionFailure("WordPressUI.ImageCache.shared should conform to GravatarImageCaching.")
+            }
+            do {
+                let _ = try await gravatar.setImage(with: fullURL,
```
Comment (etoledom) [inline WordPress/Classes/ViewRelated/Gravatar/UIImageView+Gravatar.swift]: `let` is not needed here.
Though maybe `imageView.setImage() could also have `@discardableResult`

**Review by etoledom — COMMENTED** [review]
Testing together with https://github.com/Automattic/Gravatar-SDK-iOS/pull/276 , things look good to me.
I'd say It's ready to move to formal review 👍

File: WordPress/Classes/ViewRelated/Gravatar/UIImageView+Gravatar.swift (hunk 1 above)
Comment (pinarol) [reply WordPress/Classes/ViewRelated/Gravatar/UIImageView+Gravatar.swift]: Yep let's add @discardableResult, it should be an easy change. Let me make a quick pr.

**Comment by pinarol** [comment]
Ready to review.

**Review by etoledom — APPROVED** [review]
Looks good!

**Pull request merged**